include/ggml/ggml.h

```diff
@@ -1771,6 +1771,7 @@ extern "C" {
         GGML_OPT_NO_CONTEXT,
         GGML_OPT_INVALID_WOLFE,
         GGML_OPT_FAIL,
+        GGML_OPT_CANCEL,
```

Comment:
@xaedes I've added the `GGML_OPT_CANCEL` return code and simplified the cancellation logic during optimization

Reply:
That makes a lot of sense.